Library/Homebrew/cmd/bottle.rb

```diff
@@ -183,6 +186,15 @@ def bottle_formula(f)
     relocatable = false
     skip_relocation = false
 
+    formula_source_time = f.stable.stage do
+      newest_mtime = nil
+      Pathname(Dir.pwd).find do |path|
+        newest_mtime ||= path.mtime
+        newest_mtime = path.mtime if path.mtime > newest_mtime
+      end
+      newest_mtime
+    end
```

Comment:
> If you've done brew install --build-bottle before this you'll have a cached download so shouldn't need the network.

I guess what I'm hoping to make it work without network even if I purge the bottle.

> Yup, that's another edge-case to be addressed (in a future PR if that's :ok:)

I'm OK to merge this PR as it is. But I would certainly hope to put these logics in  `formula_installer` in long term due to the reasons I already present.